WC_Layers/Source_Mesh_Reader.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 30 15:16:44 2024
+
+@author: Anupama Rajendra
+"""
+
+import h5py
+import numpy as np
+
+source_density_name = 'source_density'
+
+# Open the HDF5 files
+File_1 = h5py.File("source_mesh_1.h5m", 'r')
+File_2 = h5py.File("source_mesh_2.h5m", 'r')
+Files = [File_1, File_2]
+
+def extract_source_data(file_list):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
+    sd_list = []
+    for file in file_list:
+        # Extract data from the current file
+        tstt = file['tstt']
+        elements = tstt['elements']
+        tet_four = elements['Tet4']
+        tags = tet_four['tags']
+        sd = tags['source_density'][:]
+        sd_list.append(sd)

```

Comment:
Since you don't actually use any of these intermediate variables, I think it's probably OK do say:
```suggestion
        sd_list.append(file['tstt']['elements']['Tet4']['tags']['source_density'][:])
```

Reply:
I have a feeling that it may also be possible to use a path syntax like:
`"tsst/elements/Tet4/tags/source_density"`, but I'm not sure.

Reply:
I gave this a quick try and it didn't seem to work